### PR TITLE
AP_AHRS: move AP_NMEA_Output to AHRS frontend

### PR DESCRIPTION
**Pull request by peterbarker**
Also make it optionally-built, saving 1kB if it is removed with `HAL_NMEA_OUTPUT_ENABLED`

### CODE REVIEW

File: libraries/AP_AHRS/AP_AHRS.cpp [hunk 1]
```diff
@@ -30,6 +30,7 @@
 #include <AP_Notify/AP_Notify.h>
 #include <AP_Vehicle/AP_Vehicle_Type.h>
 #include <GCS_MAVLink/GCS.h>
+#include <AP_NMEA_Output/AP_NMEA_Output.h>
```
Comment (HefnySco) [inline libraries/AP_AHRS/AP_AHRS.cpp:33]: It is already mentioned libraries/AP_AHRS/AP_AHRS.h

Comment (peterbarker) [reply libraries/AP_AHRS/AP_AHRS.cpp]: Fixed.

File: libraries/AP_NMEA_Output/AP_NMEA_Output.cpp [hunk 2]
```diff
@@ -46,11 +45,12 @@ AP_NMEA_Output::AP_NMEA_Output()
 
 AP_NMEA_Output* AP_NMEA_Output::probe()
 {
-    if (!_singleton && AP::serialmanager().find_serial(AP_SerialManager::SerialProtocol_NMEAOutput, 0) != nullptr) {
-       _singleton = new AP_NMEA_Output();
+    AP_NMEA_Output *ret = new AP_NMEA_Output();
+    if (ret->_num_outputs == 0) {
```
Comment (tridge) [inline libraries/AP_NMEA_Output/AP_NMEA_Output.cpp:49]: check for nullptr

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_NMEA_Output/AP_NMEA_Output.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_NMEA_Output/AP_NMEA_Output.cpp]: Done

**Pull request merged**